src/big_core_input.cpp

```diff
@@ -5,316 +5,217 @@
 #include "../include/big_core_base.h"
 #include "../include/big_core_input.h"
 
-BigCoreInput::BigCoreInput()
-	:
-	lock(false)
-{}
-
 BigCoreInput::BigCoreInput(const std::string& filename)
 {
-	this->openFile(filename);
+    openFile(filename);
 }
 
 BigCoreInput::~BigCoreInput()
-{}
-
-void BigCoreInput::openFile(const std::string& filename)
 {
-	std::ifstream file(filename, std::ios_base::binary);
-	// what should be read?
-    // RV: read everything but data
-    // RV: go through the file, read numberOfImages etc., read dataOrder, dataType, DO NOT READ data, REMEMBER where data start.
-
-    // RV:
-    // 1. precis HEADER a overit
-    // 2. precist id a length
-    // 3. pokud id != 8 (DATA), precist hodnoty v chunku, NEZAPOMINAT, ze hodnoty v chunku muzou byt kratsi nez length a maji specificky datovy typ urceny pomoci id
-    // 4. skok na dalsi chunk
-    // 5. pokud neni konec souboru, opakuj od 2
+    if (file.is_open()) file.close();
 }
 
-void BigCoreInput::readFile(const std::string& filename, uint64_t bytes)
+void BigCoreInput::openFile(const std::string& filename)
 {
-    // RV:
-    // 1. volat metodu setMemorySize(bytes)
-    // 2. volat metodu openFile(filename)
-    // 3. volat metodu loadToMemory()
-
-	//this->setMemorySize(bytes);
-
-	//std::ifstream file(filename);
-	//if (!file)
-	//{
-	//	throw "Unable to open file for reading!";
-	//}
+    // open file
+    file.open(filename, std::ios_base::binary);
+    if (!file) throw "Unable to open file for reading!";
+
+    // check header
+    char buffer[CHUNK_LENGTH];
+    file.read(buffer, CHUNK_LENGTH);
+    for (size_t i = 0; i != CHUNK_LENGTH; ++i) {
+        if (buffer[i] != MAGIC[i]) throw "File format is not BIG format!";
+    }
 
-	//char buffer[CHUNK_LENGTH];
+    // clear current content
+    clear();
+    dataPosition = 0;
+
+    // load meta-data
+    while (file) {
+        CoreChunkIds id;
+        uint64_t length;
+        if (!readChunk(file, id, length) || !readData(file, id, length)) {
+            throw "Error while reading file!";
+        }
+    }
 
-	//file.read(buffer, CHUNK_LENGTH);
-	//if (!this->checkHeader(buffer))
-	//{
-	//	throw "File format is not BIG format!";
-	//}
+    // check loaded values
+    if (numberOfImages == 0) throw "Number of images in the file is zero!";
+    if (imageHeight == 0) throw "Height of images in the file is zero!";
+    if (imageWidth == 0) throw "Width of images in the file is zero!";
+    if (dataPosition == 0) throw "Data are missing in the file!";
```

Comment:
Tohle vzdycky vyhodi `"Data are missing in the file!"`. Nikde neupdatujes dataPosition